### ISSUE TITLE
Support new django meta API

### ISSUE THREAD

**Issue by sametmax** [scoursen/django-softdelete#31]
Calling delete() raises:

```
    215             self.deleted_at = timezone.now()
    216             self.save()
--> 217             for x in self._meta.get_all_related_objects():
    218                 self._do_delete(cs, x)
    219             logging.debug("FINISHED SOFT DELETING RELATED %s" % self)

AttributeError: 'Options' object has no attribute 'get_all_related_objects'
```

Because django has a new meta API in the last versions:

https://docs.djangoproject.com/en/1.10/ref/models/meta/

**Comment by Tatsh** [comment]
Just got the same error. This makes SoftDelete unusable with 1.10.

**Comment by scoursen** [comment]
This should be fixed with commit 3509a8470d55d811997ebae11219e637ffa32293

I don't have a django 1.10 project to test this against though.
